x-pack/filebeat/input/awss3/state.go

```diff
@@ -5,84 +5,52 @@
 package awss3
 
 import (
-	"fmt"
 	"time"
 )
 
 // state is used to communicate the publishing state of a s3 object
 type state struct {
-	// ID is used to identify the state in the store, and it is composed by
-	// Bucket + Key + Etag + LastModified.String(): changing this value or how it is
-	// composed will break backward compatibilities with entries already in the store.
-	ID           string    `json:"id" struct:"id"`
 	Bucket       string    `json:"bucket" struct:"bucket"`
 	Key          string    `json:"key" struct:"key"`
 	Etag         string    `json:"etag" struct:"etag"`
 	LastModified time.Time `json:"last_modified" struct:"last_modified"`
 
-	// ListPrefix is used for unique of the key in the store for awsS3WriteCommitPrefix
-	ListPrefix string `json:"list_prefix" struct:"list_prefix"`
-
 	// A state has Stored = true when all events are ACKed.
 	Stored bool `json:"stored" struct:"stored"`
-	// A state has Error = true when ProcessS3Object returned an error
-	Error bool `json:"error" struct:"error"`
+
+	// Failed is true when ProcessS3Object returned an error other than
+	// s3DownloadError.
+	// Before 8.14, this field was called "error". However, that field was
+	// set for many ephemeral reasons including client-side rate limiting
+	// (see https://github.com/elastic/beats/issues/39114). Now that we
+	// don't treat download errors as permanent, the field name was changed
+	// so that users upgrading from old versions aren't prevented from
+	// retrying old download failures.
+	Failed bool `json:"failed" struct:"failed"`
```

Comment:
Good call! I like the new name and semantics, as well as the possibility of retrying past ephemeral failures.